jib-cli/src/main/java/com/google/cloud/tools/jib/cli/Jar.java

```diff
@@ -95,13 +100,33 @@ public Integer call() {
         return 1;
       }
 
+      JarProcessorHelper.JarType jarType = JarProcessorHelper.determineJarType(jarFile);
```

Comment:
I like the direction of this. But instead of JarProcessorHelper returning a type, perhaps it could return a JarProcessor instance?

something where the code might end up looking like:
```
JarProcessor processor = JarProcessors.for(jarFile, temporaryDirectoryProvider)
containerBuilder = JarFiles.toJibContainerBuilder(processor)
...
```

Reply:
Ah that's an interesting idea. Would that mean moving all this logic to JarProcessorHelper and making it return a JarModeProcessor instance?

Reply:
I think so, you may not need the "type" enum anymore?

Reply:
I like the idea of moving all the logic out of `Jar`. I'm just not sure about having it in `JarModeProcessorHelper` since it is supposed to contain only redundant code for the JarModeProcessor? That being said, it was probably a mistake to have it contain `determineJarType` at all. How does creating a separate class for this sound?

Reply:
Yeah sure.

These are just my ideas, but I think something along these lines could work?
`JarProcessors` -> has static method for JarProcessor creation `from`
Remove `Mode` from most names since we're not exposing that anymore
`JarModeProcessor` -- rename --> `JarProcessor`
`StandardPackagedModeProcessor` -> `StandardPackagedJarProcessor`... etc?

`JarProcessorHelper` is kind of a strange name because it doesn't truly say what it is doing (which is the trap that we often fall into when writing helpers), perhaps it could be called `JarLayers` or something?

Reply:
Yes that's a good idea. `JarProcessors` may sometimes be mistaken for `JarProcessor`. I'm still struggling to find a fitting name but what do you think of `JarProcessorSelector`?

> `JarProcessorHelper` is kind of a strange name because it doesn't truly say what it is doing (which is the trap that we often fall into when writing helpers), perhaps it could be called `JarLayers` or something?
`JarLayers` is an accurate name, thank you! It describes the intent of the class pretty well.

Reply:
The name `JarProcessors` is a pretty common pattern in java, `Path` and `Paths` and we also use it all over the place. I'm not too concerned with any confusion there.

Reply:
Ok that's reassuring, thanks!